src/test/scala/mesosphere/marathon/core/task/tracker/impl/steps/UpdateTaskTrackerStepImplTest.scala

```diff
@@ -0,0 +1,201 @@
+package mesosphere.marathon.core.task.tracker.impl.steps
+
+import mesosphere.marathon.MarathonSchedulerDriverHolder
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.state.{ PathId, Timestamp }
+import mesosphere.marathon.tasks.{ TaskIdUtil, TaskTracker }
+import mesosphere.marathon.test.Mockito
+import org.apache.mesos.Protos.{ SlaveID, TaskState, TaskStatus }
+import org.apache.mesos.SchedulerDriver
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ FunSuite, GivenWhenThen, Matchers }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+class UpdateTaskTrackerStepImplTest extends FunSuite with Matchers with ScalaFutures with Mockito with GivenWhenThen {
```

Comment:
Well, testing revealed deficiencies that we were able to remove. This is not WIP, this should work in production. I should not have written the comment.

If you insist, we can create a new issue for this.